inspect_buildpack.go

```diff
@@ -0,0 +1,170 @@
+package pack
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/buildpacks/pack/internal/style"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/config"
+	"github.com/buildpacks/pack/internal/buildpack"
+	"github.com/buildpacks/pack/internal/buildpackage"
+	"github.com/buildpacks/pack/internal/dist"
+)
+
+type BuildpackInfo struct {
+	BuildpackMetadata buildpackage.Metadata
+	Buildpacks        []dist.BuildpackInfo
+	Order             dist.Order
+	BuildpackLayers   dist.BuildpackLayers
+	Location          buildpack.LocatorType
+}
+
+type InspectBuildpackOptions struct {
+	BuildpackName string
+	Daemon        bool
+	Registry      string
+}
+
+type ImgWrapper struct {
+	v1.ImageConfig
+}
+
+func (iw ImgWrapper) Label(name string) (string, error) {
+	return iw.Labels[name], nil
+}
+
+func (c *Client) InspectBuildpack(opts InspectBuildpackOptions) (*BuildpackInfo, error) {
+	locatorType, err := buildpack.GetLocatorType(opts.BuildpackName, []dist.BuildpackInfo{})
+	if err != nil {
+		return nil, err
+	}
+	var layersMd dist.BuildpackLayers
+	var buildpackMd buildpackage.Metadata
+
+	switch locatorType {
+	case buildpack.URILocator:
+		buildpackMd, layersMd, err = metadataFromArchive(c.downloader, opts.BuildpackName)
+	case buildpack.RegistryLocator:
+		buildpackMd, layersMd, err = metadataFromRegistry(c, opts.BuildpackName, opts.Registry)
+	case buildpack.PackageLocator:
+		buildpackMd, layersMd, err = metadataFromImage(c, opts.BuildpackName, opts.Daemon)
+	default:
```

Comment:
nit: order it `RegistryLocator`, `PackageLocator`, and then `URILocator`, so that you can then order the funcs in order of external and internal usage (`metadataFromRegistry`, `metadataFromImage`, `metadataFromArchive`)